build/builder.go

```diff
@@ -304,6 +304,11 @@ func (o *Options) IncrementalSkipIndexing() bool {
 		return false
 	}
 
+	// Sourcegraph specific. Ensure we have public set correctly.
+	if !rawConfigEqual(repo.RawConfig, o.RepositoryDescription.RawConfig, "public") {
+		return false
+	}
```

Comment:
This is the only thing giving me pause to merge in now. Forcing a re-index isn't great. I wonder if we have the "backpack" approach we were discussing we can be much more free to do this sort of thing?

Reply:
I see what you mean. I wonder if it is too bad though. All the active repos won't need a reindex. For the rest, there is good chance that we moslty re-index tiny stale repos.